Docs 08282024 (#45)
maffettone/erobs: docker/azure-kinect/README.md

> Build the docker / podman image

Reviewer: ```suggestion
Build the image
```

```bash
podman build -t azure-kinect:latest .
```

> Following command executes the image without opening up the network to the host network. Working with :

Reviewer: ```suggestion
The following command runs a container with this image without opening up the container network to the host network.
```

```bash

> podman run -it --privileged azure-kinect:latest /bin/bash -c "Xvfb :1 -screen 0 2560x1440x16 & . /opt/ros/humble/setup.bash && . /root/ws/install/setup.sh && ros2 launch azure_kinect_ros_driver driver.launch.py" 

Reviewer: Is the --privileged flag a requirement? If so it needs to be explained, and will not necessarily be available on all hosts. Most Podman users will not by design be allowed root access to the host machine from the container.

Author: `--privileged` flag allows the container to access all the host's devices and removes restrictions that are normally imposed on containers. This is used to grant access to the Azure Kinect camera volumes mounted on /dev/bus/usb. Instead, `--volume` flag makes the camera available to the podman container.
https://github.com/maffettone/erobs/blob/3d0d07e7881d7ecdfe12aac30b74b5d23798604e/docker/azure-kinect/README.md?plain=1#L12-L14

```

> Following command executes the image opening up a the container to the host network.

Reviewer: ```suggestion
The following command runs a container with this image using the host network as the container network.
```

```bash
podman run -it --rm --privileged --network host --ipc=host --pid=host azure-kinect:latest /bin/bash -c "Xvfb :2 -screen 0 2560x1440x16 & . /opt/ros/humble/setup.bash && . /root/ws/install/setup.sh && ros2 launch azure_kinect_ros_driver driver.launch.py depth_mode:=NFOV_UNBINNED  point_cloud_in_depth_frame:=false"
```

> A number of parameters with respect to the Kinect camera can be set via driver.launch.py. Refer [this link](https://github.com/microsoft/Azure_Kinect_ROS_Driver/blob/6ffb95a56ee175e5020b5ee5983d7230befbb176/docs/usage.md) for the complete list of options.

Reviewer: In the second command, a number of these parameters are set. They should be specified and explained above.

Author: Parameters are explained.

https://github.com/maffettone/erobs/blob/afa6a7854ecf5562b96a5e544d699b29421502cb/docker/azure-kinect/README.md?plain=1#L16
